src/errors.ml

```diff
@@ -75,14 +75,35 @@ let print ppf loc =
           Format.fprintf pp "%s\n%*s\n" line
             stop_c
             (String.make len '^')
-        else if num_lines <= 10 then
-          let lines = file_lines path ~start:start.pos_lnum ~stop:stop.pos_lnum in
-          let last_lnum = Option.map ~f:fst (List.last lines) in
-          let padding_width = Option.value_exn
-                                (Option.map ~f:String.length last_lnum) in
-          List.iter ~f:(fun (lnum, l) ->
-            Format.fprintf pp "%*s: %s\n" padding_width lnum l)
-          lines
+        else
```

Comment:
We can solve this after the other comments, but I believe that it's the point where this function has enough logic in it. It should be possible to extract a type that determines how to print a file part:

```ocaml
type display_style =
  | No_display
  | One_line of int (* line number *)
  | Block of int * int (* start, stop *)
```

Then it's possible to split the large function into one that determines the display style, and one function per display style. As a bonus, things like determining the padding size can be pulled out and factored a bit.

What do you think about that?

Reply:
I like the idea. I'll try this after I address the other comments!